symphony-bdk-core/src/main/java/com/symphony/bdk/core/util/function/Resilience4jRetryWithRecovery.java

```diff
@@ -0,0 +1,88 @@
+package com.symphony.bdk.core.util.function;
+
+import com.symphony.bdk.core.api.invoker.ApiException;
+import com.symphony.bdk.core.config.model.BdkRetryConfig;
+
+import com.symphony.bdk.core.util.BdkExponentialFunction;
+
+import io.github.resilience4j.retry.Retry;
+import io.github.resilience4j.retry.RetryConfig;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Map;
+import java.util.function.Predicate;
+
+/**
+ * This class aims to implement a retry mechanism (on top of a{@link Retry})
+ * with different recovery strategies based on predicates.
+ * @param <T> the type of the object returned by {@link #execute()}
+ */
+@Slf4j
+public class Resilience4jRetryWithRecovery<T> extends RetryWithRecovery<T> {
```

Comment:
Create a package `com.symphony.bdk.core.retry.resilience4j` and move this class there